pallet-chainlink-feed/src/lib.rs

```diff
@@ -730,7 +736,7 @@ pub mod pallet {
 
 				// update oracle rewards and try to reserve them
 				let payment = details.payment;
-				T::Currency::reserve(&T::PalletId::get().into_account(), payment).or_else(
+				T::Currency::reserve(&Self::into_account(feed_id), payment).or_else(
 					|_| -> DispatchResult {
 						// track the debt in case we cannot reserve
 						Debt::<T>::try_mutate(|debt| {
```

Comment:
As a next step, we should move Debt collecting to a per feed basis, meaning, the `FeedConfig` will take another field, `max_debt`, that tracks how much debt a feed is allowed to accumulate, and also the `debt` field for tracking it. Submitting values should only fail if `debt` > `max_debt`.